src/interface/variable_pack.hpp

```diff
@@ -307,15 +307,33 @@ class VariablePack {
   constexpr bool IsAllocated(const int /*m*/, const int /*n*/) const { return true; }
 #endif
 
+  KOKKOS_FORCEINLINE_FUNCTION
+  ParArray3D<T, VariableState> &operator()(TopologicalElement el, const int n) const {
+    assert(IsAllocated(n));
+    assert(v_((static_cast<std::size_t>(el) % 3) * dims_[3] + n).topological_type ==
+           GetTopologicalType(el));
+    return v_((static_cast<std::size_t>(el) % 3) * dims_[3] + n);
+  }
+
   KOKKOS_FORCEINLINE_FUNCTION
   ParArray3D<T, VariableState> &operator()(const int n) const {
     assert(IsAllocated(n));
+    assert(v_(n).topological_type == TopologicalType::Cell);
```

Comment:
It will, but I am not sure we should remove the assertion. My thinking was that we want people to be explicit about which type of field they are getting from a pack so that there is less opportunity for indexing errors. I am open to changing my mind though.